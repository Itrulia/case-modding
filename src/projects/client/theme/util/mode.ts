import styledTheme from "styled-theming";
import { themeFunction } from "./helpers";

export interface ModeValues<T> {
    dark: T;
}

export type ThemeMode = keyof ModeValues<any>;

export interface HasMode {
    mode: ThemeMode;
}

export const createMode = (
    key: keyof HasMode,
    obj: ModeValues<string | themeFunction>
) => styledTheme(key, obj);
