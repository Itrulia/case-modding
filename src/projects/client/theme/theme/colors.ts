import { createMode } from "../util/mode";

export const pageBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 28%, 17%)`
});

export const panelForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 5%, 93%)`
});

export const panelBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 28%, 22%)`
});

/**
 * Primary
 */
export const primaryForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 5%, 93%)`
});

export const primaryIconForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 2%, 69%)`
});

export const primaryBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 42%, 50%)`
});

export const primaryAccentBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 45%, 70%)`
});

export const primaryDimmedBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 32%, 12%)`
});

export const primarySaturatedBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 32%, 12%)`
});

export const primarySaturatedForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 15%, 63%)`
});

export const primarySaturatedIconForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 5%, 93%)`
});

/**
 * Secondary
 */
export const secondaryForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.secondaryHue}, 5%, 93%)`
});

export const secondaryIconForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.secondaryHue}, 5%, 93%)`
});

export const secondaryBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.secondaryHue}, 90%, 45%)`
});

export const secondaryAccentBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.secondaryHue}, 90%, 60%)`
});

/**
 * States
 */
export const dangerForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.dangerHue}, 5%, 93%)`
});

export const dangerBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.dangerHue}, 78%, 56%)`
});

export const warningForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.warningHue}, 5%, 93%)`
});

export const warningBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.warningHue}, 95%, 40%)`
});

export const successForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.successHue}, 5%, 93%)`
});

export const successBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.successHue}, 66%, 38%)`
});

export const disabledForeground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 5%, 93%)`
});

export const disabledBackground = createMode("mode", {
    dark: ({ theme }) => `hsl(${theme.primaryHue}, 32%, 22%)`
});
