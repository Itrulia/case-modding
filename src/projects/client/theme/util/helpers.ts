import * as styledComponents from "styled-components";
import { ThemedStyledComponentsModule } from "styled-components";
import { HasMode } from "./mode";
import { HasMood } from "./mood";

const {
    default: styled,
    css,
    keyframes,
    ThemeProvider,
    createGlobalStyle,
    ThemeContext
} = styledComponents as ThemedStyledComponentsModule<ThemeInterface>;

export interface TypographyScale {
    lineHeight: number;
    fontSize: number;
}

export interface ThemeInterface extends HasMode, HasMood {
    primaryHue: number;
    secondaryHue: number;
    // states
    dangerHue: number;
    warningHue: number;
    successHue: number;

    gridSpan: number;
}

export {
    css,
    styled,
    keyframes,
    ThemeProvider,
    createGlobalStyle,
    ThemeContext
};

export interface Props {
    theme: ThemeInterface;
}

export type themeFunction = (props: Props) => string;
