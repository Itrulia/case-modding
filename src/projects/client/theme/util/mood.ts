import styledTheme from "styled-theming";
import { themeFunction } from "./helpers";

export interface MoodValues<T> {
    playful: T;
}

export type ThemeMood = keyof MoodValues<any>;

export interface HasMood {
    mood: ThemeMood;
}

export const createMood = (
    key: keyof HasMood,
    obj: MoodValues<string | themeFunction>
) => styledTheme(key, obj);
