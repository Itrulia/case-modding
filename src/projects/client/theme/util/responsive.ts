import { css, ThemeInterface } from "./helpers";
import { ThemedCssFunction } from "styled-components";

export const createQuery = (
    minSize: number,
    maxSize?: number
): ThemedCssFunction<ThemeInterface> => (strings, ...interpolations) => {
    if (maxSize) {
        return css`
            @media (min-width: ${minSize / 16}em) and (max-width: ${maxSize /
                    16}em) {
                ${css(strings, ...interpolations)};
            }
        `;
    }

    return css`
        @media (min-width: ${minSize / 16}em) {
            ${css(strings, ...interpolations)};
        }
    `;
};
