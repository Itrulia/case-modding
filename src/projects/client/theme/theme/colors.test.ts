import { ThemeInterface } from "../util/helpers";
import { primaryBackground, primaryForeground } from "./colors";

const theme = {
    primaryHue: 211,
    secondaryHue: 52
} as ThemeInterface;

describe("primaryBackground", () => {
    it("matches dark snapshot", () => {
        const result = primaryBackground({
            theme: { ...theme, mode: "dark" }
        });
        expect(result).toMatchSnapshot(result);
    });
});

describe("primaryForeground", () => {
    it("matches dark snapshot", () => {
        const result = primaryForeground({
            theme: { ...theme, mode: "dark" }
        });
        expect(result).toMatchSnapshot(result);
    });
});
