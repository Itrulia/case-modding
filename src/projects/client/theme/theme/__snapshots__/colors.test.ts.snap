// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primaryBackground matches dark snapshot: hsl(211, 42%, 50%) 1`] = `"hsl(211, 42%, 50%)"`;

exports[`primaryForeground matches dark snapshot: hsl(211, 5%, 93%) 1`] = `"hsl(211, 5%, 93%)"`;
